includes/service-providers/class-newspack-newsletters-service-provider.php

```diff
@@ -318,4 +325,32 @@ public function send_newsletter( $post ) {
 	public function update_contact_lists( $email, $lists_to_add = [], $lists_to_remove = [] ) {
 		return new WP_Error( 'newspack_newsletters_not_implemented', __( 'Not implemented', 'newspack-newsletters' ), [ 'status' => 400 ] );
 	}
+
+	/**
+	 * Get the provider specific labels
+	 *
+	 * This allows us to make reference to provider specific features in the way the user is used to see them in the provider's UI
+	 *
+	 * @return array
+	 */
+	public static function get_labels() {
+		return [
+			'name'  => '', // The provider name.
+			'list'  => __( 'list', 'newspack-newsletters' ), // "list" in lower case singular format.
+			'lists' => __( 'lists', 'newspack-newsletters' ), // "list" in lower case plural format.
+			'List'  => __( 'List', 'newspack-newsletters' ), // "list" in uppercase case singular format.
+			'Lists' => __( 'Lists', 'newspack-newsletters' ), // "list" in uppercase case plural format.
+		];
+	}
+
+	/**
+	 * Get one specific label for the current provider
+	 *
+	 * @param string $key The label key.
+	 * @return string Empty string in case the label is not found.
+	 */
+	public static function label( $key ) {
+		$labels = static::get_labels();
+		return $labels[ $key ] ?? '';
```

Comment:
```suggestion
		return $labels[ $key ] ?? $key;
```

Since `$key` is not slugified (`Lists` is a key) I wonder if this isn't a better practice for this method.

Reply:
Oh, that should also check `isset()` to not throw warnings

Reply:
> Oh, that should also check isset() to not throw warnings

`??` does exactly that. It's a PHP 7 feature.

```PHP
return $labels[ $key ] ?? '';
return isset( $labels[ $key ] ) ? $labels[ $key ] : ''; // It's the same thing
```

That's why I don't know if it's a good idea to return `$key`... this would mean that I can pass anything and it will just return my string if the label is not found... Maybe if we had many labels this could be useful...

Reply:
TIL!

> Maybe if we had many labels this could be useful.

Sounds good, we can revisit it later if it's the case.